db.go

```diff
@@ -213,6 +215,113 @@ func (db *DB) Get(ro *ReadOptions, key []byte) ([]byte, error) {
 	return C.GoBytes(unsafe.Pointer(value), C.int(vallen)), nil
 }
 
+// GetMany returns the values associated with keys from the database.
+//
+// GetMany fills in the entries of values consistent with Get():
+// - If keys[i] exists in the database with non-zero value, values[i] holds its non-zero len []byte slice.
+// - If keys[i] does not exist in the database, values[i] == nil.
+// - If keys[i] does exist, but its value is zero-length, values[i] == []byte{}.
+// - If there's an error looking up keys[i], values[i] == nil.
+//
+// errs == nil if gets for all of the keys succeeded. Otherwise errs[i]
+// will hold the error for why get keys[i] failed. Caller can cleanly do:
+//
+// values, errs := GetMany(ro, keys)
+// if errs != nil {
+//      doErrsHandling(errs);
+// }
+//
+// The keys byte slice may be reused safely. Go takes a copy of
+// them before returning.
+func (db *DB) GetMany(ro *ReadOptions, keys [][]byte) ([][]byte, []error) {
```

Comment:
PR looks great to me now! 💥 
One thing I realized though is that interface of returning a `[]error` is pretty clunky to use as you may have seen in the usage of `if errors == nil || errors[i] == nil`

From scouring around on other libraries, it seems actually the usual signature is `GetMulti(keys ...[]byte) ([]result, error)`:
- [gorocksdb](https://godoc.org/github.com/tecbot/gorocksdb#DB.MultiGet): `MultiGet(opts *ReadOptions, keys ...[]byte) (Slices, error)`
- [redis](https://godoc.org/github.com/go-redis/redis#Client.MGet): `MGet(keys ...string) *SliceCmd` (with `SliceCmd` having `Err() error`
- GRPC (you usually do `Method(*req) (*resp, error)` even if `req` has multiple items)

So maybe a signature of `GetMany(ro *ReadOptions, keys... []byte) ([][]byte, []error)` or `GetMany(ro *ReadOptions, keys [][]byte) ([][]byte, []error)` ? (no strong opinion of `...` vs `[]`)

Sorry for late notice but looks like it would be much simpler to check error that way